src/controls/extras/tree.coffee

```diff
@@ -79,16 +80,29 @@ class Tree extends Typed
     #@inputs().jstree("get_selected", "full").map (node) ->
     #  if node.li_attr and node.li_attr.node_value and node.li_attr.relevant == 'true'
     #    node.li_attr.node_value
-    checked_required_nodes = []
-    clicked = @inputs().find('a.jstree-clicked').parent().map ->
-      node = $(this)
-      if node.attr('node_value') and node.attr('item-relevant') == 'true' and node.attr('item-required') == 'false'
-        checked_required_nodes.push(node.attr('node_value'))
-    required = @inputs().find('li[item-required=true]').map ->
-      node = $(this)
-      if node.attr('node_value') and node.attr('item-relevant') == 'true'
-        checked_required_nodes.push(node.attr('node_value'))
-    checked_required_nodes
+
+    #Get all nodes which are required, explicitely checked, or implicitely checked (i.e. all children are checked)
+    #Explicitly checked or imlicitely checked (i.e. all descendants checked, required, or irrelevant)
+    checked = @inputs().find('a.jstree-clicked').parent('[node_value][item-relevant=true][item-required=false]')
+    #Required nodes
+    required = @inputs().find('li[item-required=true][node_value][item-relevant= true]')
+    checked_required_nodes = checked.add(required)
+
+    if @simplify_output
```

Comment:
do you mean does specifying a parent node in the initial model cause all its children to be selected?  if so then yes it does.  note that simplified output is not formatted any differently, it is just a subset of the values which would be output without simplified set to true.  The lines you commented on are actually not new functionality, just a refactoring of the existing code.  then I add new logic to select out a subset of those values if simplify is enabled.